### ISSUE TITLE
Replay is broken in MAPL Develop: BaseProfiler

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#393]
Per last night's tests, MAPL `develop` seems to have broken Replay. From:
```
/gpfsm/dnb02/mathomp4/SystemTests/runs/AGCM_MAPLDEVREP/c24_O1_GOCART/CURRENT/run/1day/slurm-38682845.out
```
the traceback is:
```
pe=00000 FAIL at line=00129    BaseProfiler.F90                         <Timer Refresh should not start when empty:>
pe=00000 FAIL at line=00129    BaseProfiler.F90                         <Timer Refresh should not start when empty:>
pe=00000 FAIL at line=00129    BaseProfiler.F90                         <Timer Refresh should not start when empty:>
pe=00000 FAIL at line=00129    BaseProfiler.F90                         <Timer Refresh_self should not start when empty:>
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
GEOSgcm.x          00000000059550FD  Unknown               Unknown  Unknown
libpthread-2.11.3  00002AAAB61BA850  Unknown               Unknown  Unknown
GEOSgcm.x          00000000053A1D25  mapl_baseprofiler         160  BaseProfiler.F90
GEOSgcm.x          0000000004F0EC9F  mapl_genericmod_m        2596  MAPL_Generic.F90
GEOSgcm.x          0000000004F0EAB4  mapl_genericmod_m        2524  MAPL_Generic.F90
GEOSgcm.x          0000000004F0EAB4  mapl_genericmod_m        2524  MAPL_Generic.F90
GEOSgcm.x          0000000000598673  geos_gcmgridcompm        1705  GEOS_GcmGridComp.F90
```
I might try to figure it out myself, but I'm guessing the real @GEOS-ESM/mapl-team will get there first.

**Comment by bena-nasa** [comment]
I'm assuming this is because of the changes I made to the profilers yesterday. Probably something subtle in the Genericwrapper/genericrecord/genericrefresh. I bet this is because the record/refresh is being invoked directly in replay, not thru the wrapper and I removed the start() stop() from those assuming everything would go thru the genericwrapper. I will investigate.

**Comment by bena-nasa** [comment]
Fixed by pull request #395
